service/policy/db/key_access_server_registry.go

```diff
@@ -259,3 +261,206 @@ func (c PolicyDBClient) ListKeyAccessServerGrants(ctx context.Context, r *kasreg
 		},
 	}, nil
 }
+
+func (c PolicyDBClient) CreateKey(ctx context.Context, r *kasregistry.CreateKeyRequest) (*kasregistry.CreateKeyResponse, error) {
+	kasID := r.GetKasId()
+	key := r.GetKey()
+
+	metadataJSON, _, err := db.MarshalCreateMetadata(r.GetMetadata())
+	if err != nil {
+		return nil, err
+	}
+
+	id, err := c.Queries.createPublicKey(ctx, createPublicKeyParams{
+		KeyAccessServerID: kasID,
+		KeyID:             key.GetKid(),
+		Alg:               key.GetAlg().String(),
+		PublicKey:         key.GetPem(),
+		Metadata:          metadataJSON,
+	})
+	if err != nil {
+		return nil, db.WrapIfKnownInvalidQueryErr(err)
+	}
+
+	// Get freshly created key
+	ck, err := c.GetPublicKey(ctx, &kasregistry.GetKeyRequest{Id: id})
```

Comment:
We've been trying to avoid doing immediate GETs during CREATEs to prevent data consistency issues in a system with multiple DB replicas (planning in advance). Would you mind employing either:
- a transaction https://github.com/opentdf/platform/blob/9e7d5b972f9e55752b79e3191b7625b7389b1b2d/service/policy/db/policy.go#L37
- populating everything in the response from the `RETURNING` SQL and request?

Reply:
wrapped it in a transaction